File: Ytnest-Backend/src/main/java/com/YTnest/springboot/model/Record.java
package com.YTnest.springboot.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
//import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.*;

import lombok.Data;

@Data
@Entity
@Table(name = "record")

public class Record {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "record_id")
    private Long recordId;

    @Column(name = "record_description")
    private String recordDescription;

    @Column(name = "status")
    private String status;

    @Column(name = "score")
    private int score;

    @JsonIgnore
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "record_id")
    private List<RecordPenalty> recordPenalty; 
      
    @OneToOne
    @JoinColumn(name = "student_id")
    @JsonIgnoreProperties("record") // Evita la serialización en bucle
    private Student student;

    @JsonProperty("student_id")
    public Long getStudentId() {
        if (student != null) {
            return student.getStudent_id();
        }
        return null;
    }
}

File: YTnest-Frontend/src/app/classes/record.ts
import { RecordPenalty } from "./record-penalty";
import { Student } from "./student";

export class studentRecord {
  recordId?: number;
  student_id?: number;
  recordDescription?: string;
  score?: number;
  status?: string;
  recordPenalty?: RecordPenalty | null;
}

export class RECORD{
  recordId?: number;
  recordDescription?: string;
  status?: string;
  score?: number;
  recordPenalty?: RecordPenalty | null;
  student?: Student;
}
